overlord/devicestate/handlers_gadget.go

```diff
@@ -210,11 +211,36 @@ func (m *DeviceManager) doUpdateGadgetAssets(t *state.Task, _ *tomb.Tomb) error
 	return snapstate.FinishTaskWithRestart(t, state.DoneStatus, restart.RestartSystem, nil)
 }
 
-func (m *DeviceManager) updateGadgetCommandLine(t *state.Task, st *state.State, isUndo bool) (updated bool, err error) {
-	snapsup, err := snapstate.TaskSnapSetup(t)
+func buildDynamicKernelCommandLine(st *state.State) (string, error) {
+	var cmdlineDyn, cmdlineDanger string
+	tr := config.NewTransaction(st)
+	if err := tr.Get("core", "system.boot.cmdline-extra",
```

Comment:
we will need to deal with dangerous too, no?

Reply:
I'm looking at this option for all cases and using -dangerous only for dangerous grade and if this one is empty, see code below. Let me know if that is not the expected behavior.